packages/dev/gui/src/2D/controls/inputTextArea.ts

```diff
@@ -573,7 +576,7 @@ export class InputTextArea extends InputText {
 
     protected _breakLines(refWidth: number, context: ICanvasRenderingContext): object[] {
         const lines: { text: string; width: number; lineEnding: string }[] = [];
-        const _lines = this.text.split("\n");
+        const _lines = (this.text || this.placeholderText).split("\n");

```

Comment:
HTML Placeholder works a bit differently. If you add a placeholder, it is set as a background of the input, and if you focus on the element, you can instantly overwrite it.
This implementation does work this way, apart from a single issue - when focusing on the element, the cursor is at the end:

![image](https://github.com/BabylonJS/Babylon.js/assets/1381702/edf4eb04-c9b1-4051-8e40-6ae681f9bf85)

PG is this - https://playground.babylonjs.com/?snapshot=refs/pull/14587/merge#NVAEWD#364

Now - this is more of a question - is it possible to set the cursor to the beginning of the element?

Reply:
@RaananW great point, I've updated the code to reset the cursor back to 0 when it's focused and displaying a placeholder 🙂